nicegui/run.py

```diff
@@ -33,6 +33,7 @@ def run(*,
         uvicorn_reload_excludes: str = '.*, .py[cod], .sw.*, ~*',
         exclude: str = '',
         tailwind: bool = True,
+        custom_ctrl_c_handler = True,
```

Comment:
Why do we need a parameter in ui.run to enable/disable the behaviour? Are there any side-effects of setting the console ctrl handler?

Reply:
There can only be one handler, I suppose one side effect would be if it were to conflict with any other library (if that library were to make use of SetConsoleCtrlHandler as well).

Likewise SIGTERM is a forced exit. so say if you had:
```py
import atexit
atexit.register(lambda: print("exit"))
```
"exit" would never print.  I had a go using `exit()` (which would be a more graceful way of quitting) instead of `os.kill(os.getpid(), 15)` but it seems to always fail once inside of `ui.run()` (I suppose this is uvicorns fault).

Reply:
I've now added a manual call to `atexit._run_exitfuncs()` to resolve at least one of the things mentioned in above comment.

Let me know if you want me to remove the `custom_ctrl_c_handler` parameter (though, as my local python scripts also messes with SetConsoleCtrlHandler I would prefer to be able to toggle this function off).

Reply:
I see. Thanks for elaborating @ItsCubeTime. 

> this is also normal behavior on Windows when sending ctrl C to a regular python program

I assume "simple" programs can be aborted with ctrl-c on Windows, too. ChatGPT said:

> This approach seems to be a workaround for the problem, but keep in mind that directly calling private functions (like _run_exitfuncs) can be risky because their behavior is not part of the public API and may change in future versions of Python. It's usually better to structure your program so that it can be interrupted by a KeyboardInterrupt exception, which is the normal way for Ctrl+C to terminate a Python program. If that's not possible for some reason, this workaround might be a viable alternative.